internal/manifests/collector/poddisruptionbudget.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"github.com/go-logr/logr"
+	policyV1 "k8s.io/api/policy/v1"
+	policyV1Beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/internal/config"
+	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/autodetect"
+)
+
+func PodDisruptionBudget(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) client.Object {
```

Comment:
can we add a comment here linking to the deprecation docs so we know when we can remove the dual state? https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125

Reply:
I am fine bumping the lowest support k8s version to 1.21. 

It seems that the oldest supported k8s version by a cloud provider out there is 1.23. I will start a GH issue to discuss it.

Reply:
If the lowest version is 1.21, we don't need to autodetect versions for pdb, and if the lowest version would be 1.23, the same applies for autoscaling (autoscaling/v2 was introduced on 1.23).
I don't have problems to update my PR if you have consensus about the lowest version change